File: 08-streaming/8.1-over-tcp/src/main/scala/x/NetworkWordCount.scala
package x


import org.apache.spark.SparkConf
import org.apache.spark.streaming.StreamingContext
import org.apache.spark.streaming.Seconds
import org.apache.spark.storage.StorageLevel

/*
$   ~/spark/bin/spark-submit  --master local[2]   --driver-class-path logging/  --class x.NetworkWordCount target/scala-2.11/over-tcp_2.11-1.0.jar
 */


object NetworkWordCount {
  def main(args: Array[String]) {

    val sparkConf = new SparkConf().setAppName("NetworkWordCount")

    val ssc = new StreamingContext(sparkConf, Seconds(10))

    // listen on port 9999, only cache it local memory
    val lines = ssc.socketTextStream("localhost", 9999, StorageLevel.MEMORY_ONLY)
    lines.print

    val words = lines.flatMap(_.split(" "))
    val pairs = words.map(w => (w, 1))
    val wordCounts = pairs.reduceByKey(_ + _) 
    wordCounts.print()


    // kick off the processing
    ssc.start()
    ssc.awaitTermination()
  }
}

File: 08-streaming/8.1-over-tcp/python/networkwordcount.py
from pyspark import SparkContext
from pyspark.streaming import StreamingContext
from operator import add

# Create a local StreamingContext with two working thread and batch interval of 1 second
sc = SparkContext("local[2]", "NetworkWordCount")

ssc = StreamingContext(sc, 10)

## TODO-1: listen on port 10000, only cache it local memory
lines = ssc.socketTextStream("localhost", ???)
lines.pprint()

words = lines.flatMap(lambda x: x.split(" "))
pairs = words.map(lambda w: (w, 1))
wordCounts = pairs.reduceByKey(add) 
wordCounts.pprint()


ssc.start()             # Start the computation
ssc.awaitTermination()  # Wait for the computation to terminate
